src/panels/lovelace/elements/hui-conditional-element.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Condition,
+  checkConditionsMet,
+  validateConditionalConfig,
+} from "../../lovelace/common/validate-condition";
+import { createConfiguredHuiElement } from "../../lovelace/cards/picture-elements/create-configured-hui-element";
+
+import { LovelaceElement, LovelaceElementConfig } from "./types";
+import { HomeAssistant } from "../../../types";
+
+interface Config extends LovelaceElementConfig {
+  conditions: Condition[];
+  elements: LovelaceElementConfig[];
+}
+
+class HuiConditionalElement extends HTMLElement implements LovelaceElement {
+  public _hass?: HomeAssistant;
+  private _config?: Config;
+  private _elements: LovelaceElement[] = [];
+
+  public setConfig(config: Config): void {
+    if (
+      !config.conditions ||
+      !Array.isArray(config.conditions) ||
+      !config.elements ||
+      !Array.isArray(config.elements) ||
+      !validateConditionalConfig(config.conditions)
+    ) {
+      throw new Error("Error in card configuration.");
+    }
+
+    this._config = config;
+
+    this._config.elements.map((elementConfig: LovelaceElementConfig) => {
+      this._elements.push(
+        createConfiguredHuiElement(elementConfig, this._hass!)
+      );
+    });
+
+    if (this._hass) {
+      this.hass = this._hass;
```

Comment:
You're doing this here and in connectedCallback. It's confusing. I think that we should instead just have a `render` method that does the right thing ?

Reply:
It's needed because config can be called even after component was fully loaded.

Reply:
And render won't help since we're adding the elements to the parent to maintain positioning as regular `picture-elements` elements...